src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1064,7 +1076,7 @@ private Task WriteSuffix()
 
             if (!HasFlushedHeaders)
             {
-                _requestProcessingStatus = RequestProcessingStatus.HeadersFlushed;
+                _requestProcessingStatus = RequestProcessingStatus.ResponseCompleted;
```

Comment:
Did this incorrectly setting just HeadersFlushed instead of ResponseCompleted result in any bad behavior prior to this change?

Reply:
Never mind. Now I see this state didn't exist before. That makes more sense.

Reply:
ResponseCompleted is new. This flush operation flushes and completes so I had it skip the flush state.